File: python/0035.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if target in nums:
            return nums.index(target)
        else:
            for i in range(len(nums)):
                if nums[i]>target:
                    return i
            return len(nums)

File: rust/src/bin/0035.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut l = 0;
        let a = nums;
        let x = target;
        let mut r = a.len();
        while l < r {
            let mid = (l + r) >> 1;
            if a[mid] < x {
                l = mid + 1;
            } else {
                r = mid;
            }
        }
        l as i32
    }
}